app/controllers/setting_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SettingController < ApplicationController
+
+  def show
+    @apis = Setting.find_by(user_id: current_user.id).api
+    @tags = Setting.find_by(user_id: current_user.id).tags
+    @categories = News.distinct.pluck(:category)
+  end
+  
+  def update
+    current_api = []
```

Comment:
this method is too big. You can update setting much easier just using form that is sent from settings page. 

if params was like:
```
setting_params: {
  api: ["mediastack", "currentsapi"],
  tags: ["sport", "news"]
```
you would just update your setting instance with `setting.update(setting_params)`